### PR TITLE
Content group horizontal content model

### PR DESCRIPTION
**Pull request by mfearer94**
I added the content model to the "content group - horizontal" component and added an anchor link that goes to the content model.

### Changelog

**New**

- Content model
- Anchor link

### CODE REVIEW

**Review by JennySanchez — APPROVED** [review]
Looks good to me.

File: src/pages/components/content-group-horizontal.mdx [hunk 1]
```diff
@@ -11,15 +11,26 @@ import ComponentFeedback from 'components/ComponentFeedback';
...
 <AnchorLink>Feedback</AnchorLink>
 
 </AnchorLinks>
 
+## Content guidance
+
+| Field | Field type | Required   | Cardinality | Max character size (English / translated) |
+| -------- | --------- | ---------- | ---------- | ----------|
+| Headline | Short copy | Yes | 1 | XXS (40 / 55)|
+| TOC headline override | Short copy | No | 1 | Mini (25 / 35)|
```
Comment (photodow) [inline src/pages/components/content-group-horizontal.mdx:26]: What is this line referring to?

File: src/pages/components/content-group-horizontal.mdx [hunk 2]
```diff
@@ -11,15 +11,26 @@ import ComponentFeedback from 'components/ComponentFeedback';
...
 <AnchorLink>Development documentation</AnchorLink>
 <AnchorLink>Feedback</AnchorLink>
 
 </AnchorLinks>
 
+## Content guidance
+
+| Field | Field type | Required   | Cardinality | Max character size (English / translated) |
+| -------- | --------- | ---------- | ---------- | ----------|
+| Headline | Short copy | Yes | 1 | XXS (40 / 55)|
```
Comment (photodow) [inline src/pages/components/content-group-horizontal.mdx:25]: As much as it pains me to say, the heading here is using Content block's heading. For what it's worth.

**Review by photodow — COMMENTED** [review]
Need clarification on Table of Contents here.

File: src/pages/components/content-group-horizontal.mdx (hunk 1 above)
Comment (mfearer94) [reply src/pages/components/content-group-horizontal.mdx:26]: Do you mean the "TOC headline override"? If so, that field came from the need to change just the table of contents headline. Some users thought the headline could be too long for the table of contents so they wanted a field added where they could override the default headline auto-population. @photodow

File: src/pages/components/content-group-horizontal.mdx (hunk 2 above)
Comment (mfearer94) [reply src/pages/components/content-group-horizontal.mdx:25]: Hahaha we will need to discuss this more ;)

**Pull request merged**